### PR TITLE
Bot treats all pokemon as VIP fix

### PR DESCRIPTION
**Pull request by Gobberwart**
## Short Description:

Bot treats all pokemon as VIP. Additionally, default rules always return true if not configured. For example:

```
    "catch": {
      "any": {"catch_above_cp": 500, "catch_above_iv": 0.65, "logic": "or"}
    },
```

Given example pokemon with cp 30, iv 0.2 and ncp 0.15, the bot would still always catch it because no "catch_above_ncp" rule is defined. This is incorrect behaviour, should return false if no condition matches or condition doesn't exist.

## Fixes/Resolves/Closes (please use correct syntax):
- #5457 
- #5476 
- #5491

### CODE REVIEW

**Comment by Gobberwart** [comment]
Replaces #5479

**Comment by Anakin5** [comment]
`catch_results` are initialized to False. So if the don't have any rule on `cp`, `catch_results["cp"]` will be False, right ? 
So with logic at `and`, no Pokemon will get caught, right ?

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py [hunk 1]
```diff
@@ -44,6 +44,7 @@
     'andor': lambda x, y, z: x and y or z
 }
 
+DEBUG_ON = False
```
Comment (alexyaoyang) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: This is probably just a small issue, but should we introduce another debug flag here? We already have a debug flag in config now.

Comment (Gobberwart) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: this flag only applies to this rather than using global debug which generates a lot of garbage

**Comment by Gobberwart** [comment]
@Anakin5 That's correct as it stands... The bot really shouldn't use a rule if it's not defined in config. Let me take a look.

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (alexyaoyang) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: Hmm ok noted

**Comment by Gobberwart** [comment]
@Anakin5 Added some logic to handle different default values based on catch logic. It's mucky but should account for all configurations.

Having said that, the current LOGIC_TO_FUNCTION only works if there are exactly three values. Would be much better if we could only compare what's actually set in config and ignore whatever is not set. 

Leaving as-is for now so we can make the bot work properly again.

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (Gobberwart) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: It's not unusual. A lot of other code has the same thing. It's actually really handy when debugging a single module.

**Comment by Aelfwin1996** [comment]
please merge these changes to master branch. I have updated every day since these commits but still have this issue on master branch

**Pull request merged**